aws-xray-agent/src/test/java/com/amazonaws/xray/agent/runtime/config/XRaySDKConfigurationTest.java

```diff
@@ -454,4 +454,13 @@ public void testContextPropagation() {
 
         Assert.assertTrue(AWSXRay.getGlobalRecorder().getSegmentContextResolverChain().resolve() instanceof ThreadLocalSegmentContext);
     }
+
+    @Test
+    public void testLazyLoadTraceIdInjection() {
```

Comment:
How does this test work? Should we have a spring boot integration test?

Reply:
The test works by taking a vanilla recorder and verifying that the lazy load method does in fact add the trace ID Injecting listeners to it. But I'd agree this fix is not very suited to be tested with a unit test. 

Would Spring Boot integ tests have to be a full smoke test with container images and such or could we add them in the same format as existing integ tests? Given the trouble the agent has had with Spring Boot in the past it would be very nice to get such tests in place somehow to cover other functions of the agent too.

Reply:
I think it might be possible to use test-sets to have a classpath for a specific test if the integration tests would work with that

https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/otlp/trace/build.gradle#L13

Reply:
I see, so is there a way where I could define a test class path with one of these test-sets which is not visible to the agent at premain, but is used by the application? I'm just trying to reach a solution that might simulate the Spring Boot class loader's uniqueness, but am not sure it's possible without committing to a smoke test.

Reply:
Ah - yeah that's a good point, I forgot that spring boot really only kicks in when packaging a final distribution jar. So smoke test does sound like what is in order :(